### PR TITLE
Return error if mount failed

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#3599

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -329,5 +329,5 @@ func MountWithTimeout(mounter mount.Interface, source string, target string, fst
 	if !mountComplete {
 		return errors.Wrapf(err, "mounting %v share %v on %v timed out", fstype, source, target)
 	}
-	return nil
+	return err
```
Comment (shuo-wu) [inline util/util.go:332]: Can there be a case that `mountComplete` is true but the error is not empty?

Comment (derekbit) [reply util/util.go:332]: Yeah, for example, giving a wrong address.

Comment (shuo-wu) [reply util/util.go:332]: Does this mean that the device will be mounted at the wrong address? If YES, does this function need to revert the change (do unmount) when there is an error?

Comment (derekbit) [reply util/util.go:332]: > Does this mean that the device will be mounted at the wrong address?

I mean a not existing remote address. The error will be returned.

But you're right. For a existing but wrong address, we should unmount it. I will open another ticket for this issue.

**Review by shuo-wu — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@Mergifyio backport v1.4.x v1.3.x

**Comment by innobead** [comment]
https://github.com/Mergifyio backport v1.4.x v1.3.x

**Pull request merged**